src/sentry/ingest/ingest_consumer.py

```diff
@@ -270,10 +271,30 @@ def _load_event(
     def dispatch_task(cache_key: str) -> None:
         if attachments:
             with sentry_sdk.start_span(op="ingest_consumer.set_attachment_cache"):
-                attachment_objects = [
-                    CachedAttachment(type=attachment.pop("attachment_type"), **attachment)
-                    for attachment in attachments
-                ]
+                attachment_objects = []
+                for attachment in attachments:
+                    attachment_type = attachment.pop("attachment_type")
+                    if attachment_type == "event.view_hierarchy" and has_proguard_file(data):
+                        view_hierarchy = attachment_cache.get_from_chunks(
+                            key=cache_key, type=attachment_type, **attachment
+                        )
+                        view_hierarchy = json.loads(attachment_cache.get_data(view_hierarchy))
+                        view_hierarchy = deobfuscate_view_hierarchy(data, project, view_hierarchy)
```

Comment:
Ah, I thought @untitaker's comment was r.e. standalone attachments, but I can see that this operation could be resource intensive and handling this out of the normal flow would be ideal. It would be great if I could have help pairing on that because I'm not too familiar with our best practices for setting up these tasks

There are two files that are used here: the proguard file for deobfuscating, and the view hierarchy. The proguard file can be unbounded and the SDK team has reported seeing instances between 50-200mb. As for the view hierarchies, these are newly collected so we don't have customer data for them, but our sample apps are sending ~1-5kb per view hierarchy for this Android-specific case. Flutter is not relevant here in this deobfuscation, but as a reference for size those go up to ~84kb where the size difference is due to how Flutter works.

So the biggest resource consumption here comes from loading up the proguard file

Reply:
After looking at the code more closely, I think that for inline attachments, we might want to hook into the [`JavaPlugin`](https://github.com/getsentry/sentry/blob/a00103baf3cc8941c6b5bf8b31f7bfaccb14523c/src/sentry/lang/java/plugin.py#L130), which is called from [`process_event`](https://github.com/getsentry/sentry/blob/a00103baf3cc8941c6b5bf8b31f7bfaccb14523c/src/sentry/tasks/store.py#L374) and already deals with proguard files.

I am not sure how easy it is to access attachments from there though, you might have to add a new `get_attachment_processors` method to the plugin, and call it from the same place as `get_event_preprocessors`.